gcs/upload.py

```diff
@@ -280,6 +282,137 @@ def init_write_object_grpc(cls, db, request_iterator, context):
             upload.metadata.metadata["x_emulator_no_md5"] = "true"
         return upload, is_resumable
 
+    @classmethod
+    def process_bidi_write_object_grpc(cls, db, request_iterator, context):
```

Comment:
optional: this is a rather long function, consider breaking it down, maybe split up the function to process the first message?

Reply:
Agreed, I'm still trying to see how this can be split up without affecting the generator